pocs/tests/conftest.py

```diff
@@ -22,19 +23,38 @@ def images_dir(tmpdir_factory):
 
 
 @pytest.fixture(scope='function')
-def config(images_dir):
+def config(images_dir, messaging_ports):
     pocs.base.reset_global_config()
 
     global _one_time_config
     if not _one_time_config:
         _one_time_config = load_config(ignore_local=True, simulator=['all'])
+        # Set several fields to fixed values.
         _one_time_config['db']['name'] = 'panoptes_testing'
         _one_time_config['name'] = 'PAN000'  # Make sure always testing with PAN000
         _one_time_config['scheduler']['fields_file'] = 'simulator.yaml'
 
-    _one_time_config['directories']['images'] = images_dir
+    # Make a copy before we modify based on test fixtures.
+    result = copy.deepcopy(_one_time_config)
 
-    return copy.deepcopy(_one_time_config)
+    # We allow for each test to have its own images directory, and thus
+    # to not conflict with each other.
+    result['directories']['images'] = images_dir
+
+    # For now (October 2018), POCS assumes that the pub and sub ports are
+    # sequential. Make sure that is what the test fixtures have in them.
+    # TODO(jamessynge): Remove this once pocs.yaml (or messaging.yaml?) explicitly
+    # lists the ports to be used.
+    assert messaging_ports['cmd_ports'][0] == (messaging_ports['cmd_ports'][1] - 1)
```

Comment:
Does this make more sense in the `messaging_ports` fixture?

Reply:
Yes, except that the point here is to ensure that the messaging_ports has provided the correct pairs. The fix to do so was in another PR, hence that is why I put it here, and also because I wanted the enforcement closer to where the problem arises... still not here, but by the time we're in POCS we don't see that there are pairs of ports.

Reply:
FYI, I plan to get the the point where pocs.yaml actually has the pair of ports listed, rather than implicit, which will eliminate the need for these asserts.